server/controllers/finance/reports/cashflow/index.js

```diff
@@ -753,3 +169,241 @@ function reportByService(req, res, next) {
     .catch(next)
     .done();
 }
+
+/**
+ * This function get periodic balances by transaction type
+ * reporting transaction type balance detailled by accounts
+ * with their balance for each transaction type
+ */
+function report(req, res, next) {
+  let serviceReport;
+  const dateFrom = new Date(req.query.dateFrom);
+  const dateTo = new Date(req.query.dateTo);
+  const options = _.clone(req.query);
+  const data = {};
+
+  // convert cashboxesIds parameters in array format ['', '', ...]
+  // this parameter can be sent as a string or an array we force the conversion into an array
+  const cashboxesIds = _.values(req.query.cashboxesIds);
+
+  _.extend(options, { orientation : 'landscape' });
+
+  // catch missing required parameters
+  if (!dateFrom || !dateTo || !cashboxesIds.length) {
+    throw new BadRequest('ERRORS.BAD_REQUEST');
```

Comment:
In the future, it would be good to include a text for our logs describing why the error happened.  I think `BadRequest` accepts multiple parameters:
```js
throw new BadRequest('ERRORS.BAD_REQUEST', 'You must have a start and end date defined and provide at least one cashbox id.');
```
This is purely to help us debug any errors - I don't think it is shown to the user.

Reply:
The idea here is to throw an error message if `dateFrom`, `dateTo` or `cashboxesIds` are missing.